crates/orderbook/src/orderbook.rs

```diff
@@ -194,6 +209,22 @@ impl Orderbook {
         Ok(solvable_orders)
     }
 
+    pub async fn get_auction(&self) -> Result<Auction> {
+        let last_handled_block = self.event_updater.last_handled_block().await;
+        let solvable_orders = self.get_solvable_orders().await?;
```

Comment:
It would be nice to synchronize last_handled_block with solvable orders. If you look in SolvableOrders.rs, it gets from the database the orders and last_settlement_block in one transaction. We can extend that to also include the last event block from EventStoring::last_event_block.
The way it is currently implemented the orders and event block can get out of sync and getting the last handled block from the new EventUpdating trait has to go through a mutex which feels weird.
Implementing this might be somewhat annoying because we have to share the last event query with both places so has to be a pub str somewhere.

Reply:
Note that, AFAIU, `last_handled_block != last_event_block`. Specifically, the last handled block is the last block that was used to request events from the node:
```rust
self.last_handled_block = Some(range.end().to_u64());
```

While `last_event_block` is:
```sql
SELECT GREATEST(
  (SELECT COALESCE(MAX(block_number), 0) FROM trades),
  (SELECT COALESCE(MAX(block_number), 0) FROM settlements),
  (SELECT COALESCE(MAX(block_number), 0) FROM invalidations),
  (SELECT COALESCE(MAX(block_number) , 0) FROM presignature_events)
);
```

This means that even if there are no events which we process for a long time, we would still have `last_handled_block ~= "latest"` (which is what I intended from the auction end point, specifically because prices will be around this block number and if we are too far in the past then the prices are not necessarily valid).

We could also fetch the last event block with the orders (instead of using just the last settlement block as it does now) for the case where the event updater hasn't updated at all yet, which would be slightly more correct. I briefly touched on this in the PR description:

> In case the event handler has not processed any blocks - it returns the block number of the last settlement (as the `v2/solvable_orders` endpoint). This should only happen right on startup so I'm not particularly worried about optimizing the case where no events have been processed.

Reply:
Your explanation makes sense. I missed the distinction.